app/controllers/trainees/contact_details_controller.rb

```diff
@@ -26,5 +30,20 @@ def contact_details_params
         *ContactDetail::FIELDS,
       )
     end
+
+    def contact_details_attributes
+      ContactDetails::Utilities::AddressAttributes.new(contact_details_params).attributes
```

Comment:
I suggest deleting `ContactDetails::Utilities::AddressAttributes` (it's not service object anyway) and put all the logic into the `ContactDetail` model.

Reply:
@darokel ^

Reply:
I suggest keeping it out. The `ContactDetail` is really just a validator. Its job is to validate the attributes currently on a given trainee. Manipulating the attributes should be handled outside to keep this class lightweight and simple.

Reply:
@darokel Check out the last commit. @aliuk2012 was helping me fix a frontend issue I was having earlier and then we had a chat about this particular area of code. We ended up pairing for a bit to see if there was another way. Let us know what you think.

Reply:
It's out of the validator so I'm happy with that. Tacking it on the trainee and having to execute the logic via an explicit method makes sense. I've already approved the PR so will leave it at that 👍